# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.h
# Commit: 16d84439756f
# Full Hash: 16d84439756fae6ab67f7e874dd61d03d3b28905
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - WebXR compositing fixes. r=imanol
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78117
# ==============================================================================

diff -r bbfe23c61add -r 16d84439756f dom/canvas/ClientWebGLContext.h
--- a/dom/canvas/ClientWebGLContext.h	Mon Jun 08 20:34:15 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.h	Mon Jun 08 20:36:14 2020 +0000
@@ -974,7 +974,6 @@
   layers::LayersBackend GetCompositorBackendType() const;
 
   bool mCapturedFrameInvalidated = false;
-  UniquePtr<layers::SurfaceDescriptor> mFrontBufferDesc;
 
   // -------------------------------------------------------------------------
   // WebGLRenderingContext Basic Properties and Methods
@@ -995,8 +994,9 @@
   }
   void GetContextAttributes(dom::Nullable<dom::WebGLContextAttributes>& retval);
 
-  void Present();
-  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(layers::TextureType);
+  void Present(WebGLFramebufferJS*, layers::TextureType);
+  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(WebGLFramebufferJS*,
+                                                  layers::TextureType);
   RefPtr<gfx::SourceSurface> GetFrontBufferSnapshot() override;
 
  private: